src/Sulu/Bundle/AdminBundle/Metadata/FormMetadata/OnlyKeysVisistor.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\AdminBundle\Metadata\FormMetadata;
+
+use Sulu\Bundle\AdminBundle\Metadata\SchemaMetadata\SchemaMetadata;
+
+class OnlyKeysVisistor implements TypedFormMetadataVisitorInterface
+{
```

Comment:
This feels a little bit strange for me doing this as a Visitor. But I'm not familiar enough with the metadata process here, but for me it more would feel like a new serialization group. But I not even know how the seralization process works here.

/cc @wachterjohannes what do you think is best to achieve this kind of thing?

Reply:
I completely agree that doing it with a serialization group would probably be the best idea. The optimal solution would just to add another endpoint and only load the information needed. (Right now it still loads the entire structure but only outputs a subsection.)

However, I don't know how Sulu handles Serialization groups. That's why I chose this approach.

Reply:
@alexander-schranz @mamazu i would also have to take a deeper look at that. But in the first look we use the "normal" serialization process (in our case JMS) to output the metadata (see https://github.com/sulu/sulu/blob/2.5/src/Sulu/Bundle/AdminBundle/Controller/AdminController.php#L335) so groups would be quite easy to add.

Reply:
Yeah then I think the new query parameter (?keys-only=true) should add a specific serialization group (`admin_form_metadata_keys_only`) which gets the the `@Exclude` on the items so only the keys are returned.